### PR TITLE
Enable scan ConstantPool Objects during Scavenge

### PR DESCRIPTION
**Pull request by LinHu2016**
- new global variable enableScanConstantPoolObjectInScavenge
	(default:false)

Signed-off-by: Lin Hu <linhu@ca.ibm.com>

### CODE REVIEW

**Comment by LinHu2016** [comment]
@dmitripivkine please review the change, Thanks

File: gc/base/GCExtensionsBase.hpp [hunk 1]
```diff
@@ -499,6 +499,7 @@ class MM_GCExtensionsBase : public MM_BaseVirtual {
 	};
 
 	HeapInitializationSplitHeapSection splitHeapSection; /**< Split Heap section to be requested */
+	bool enableScanConstantPoolObjectInScavenge; /**<  set true if need to scan ConstantPool Objects during Scavenge, the default is false (set true if Java 11 and above) */
```
Comment (dmitripivkine) [inline gc/base/GCExtensionsBase.hpp]: Why did you add it here but not in `MM_GCExtensions`? This is Java related content should not be in OMR
You don't need change OMR code at all

Comment (dmitripivkine) [reply gc/base/GCExtensionsBase.hpp]: Alternatively this flag might be put in `MM_ScavengerDelegate` so name can be reduced to `enableScanConstantPoolObject`. If we don't need command line option and can handle RootScannerReadBarrierVerivier (add ScavengerDelegate there) it is probably preferable solution

**Comment by LinHu2016** [comment]
move variable to openj9 since it is only related with java
